### PR TITLE
export verify_message and incoming_error

### PR DESCRIPTION
**Pull request by ssiloti**
Also move these functions to a new msg.cpp. They are useful for plugins
which implement DHT extensions.

### CODE REVIEW

**Comment by arvidn** [comment]
I actually have on my list to add a template interface to verify_message. It's too easy to not provide enough slots in the array, or have the array size mismatch what's passed in.

(I was waiting for your IPv6 patch to land first).

**Comment by ssiloti** [comment]
Oh yeah, I was thinking the same thing the other day (great minds) but forgot about it. I'll withdraw this for now.

**Comment by arvidn** [comment]
oh.. well, this still seems like a step in the right direction though

**Comment by arvidn** [comment]
to export it, I would imagine the template would just be in a header file and it would forward to the current function, which probably would be moved into the libtorrent::aux namespace (but still be exported)

**Comment by ssiloti** [comment]
Exactly what I had in mind.

File: include/libtorrent/kademlia/msg.hpp [hunk 1]
```diff
@@ -30,16 +30,17 @@ POSSIBILITY OF SUCH DAMAGE.
...
 #include <string>
-#include <libtorrent/kademlia/node_id.hpp>
-#include <boost/asio/ip/udp.hpp>
+#include "libtorrent/socket.hpp"
+#include "libtorrent/kademlia/node_id.hpp"
 
 namespace libtorrent {
 
 struct bdecode_node;
+struct entry;
```
Comment (arvidn) [inline include/libtorrent/kademlia/msg.hpp]: class entry;

**Pull request merged**